src/js/components/MenuList.js

```diff
@@ -0,0 +1,131 @@
+  const $menuForm = $("#espresso-menu-form");
+  const $menuInput = $(".input-field");
+  const $menuCount = $(".menu-count");
+  const $menuList = $("#espresso-menu-list");
+
+  let menuData = [];
+
+  const countMenu = () => {
+    // 총 메뉴 갯수를 count하여 상단에 보여준다
+    const count = $menuList.querySelectorAll("li").length;
+    $menuCount.innerHTML = `총 ${count} 개`;
+  };
+
+  const drawMenu = (data) => {
+    const menuItem = document.createElement("li");
+    menuItem.className = "menu-list-item d-flex items-center py-2";
+    menuItem.id = data.id;
+    const menuItemName = document.createElement("span");
+    // 품절 상태인 경우를 보여줄 수 있게, 품절 버튼을 추가하고 sold-out class를 추가하여 상태를 변경한다
+    menuItemName.className = data.isSoldOut
+      ? "w-100 pl-2 menu-name sold-out"
+      : "w-100 pl-2 menu-name";
+    menuItemName.appendChild(document.createTextNode(data.name));
+    const menuItemSoldOutButton = document.createElement("button");
+    menuItemSoldOutButton.className =
+      "bg-gray-50 text-gray-500 text-sm mr-1 menu-sold-out-button";
+    menuItemSoldOutButton.onclick = (e) => soldOutMenu(e);
+    menuItemSoldOutButton.appendChild(document.createTextNode("품절"));
+    const menuItemUpdateButton = document.createElement("button");
+    menuItemUpdateButton.className =
+      "bg-gray-50 text-gray-500 text-sm mr-1 menu-edit-button";
+    // 메뉴의 수정 버튼을 눌러 메뉴 이름을 수정할 수 있다
+    menuItemUpdateButton.onclick = (e) => updateMenu(e);
+    menuItemUpdateButton.appendChild(document.createTextNode("수정"));
+    const menuItemDeleteButton = document.createElement("button");
+    menuItemDeleteButton.className =
+      "bg-gray-50 text-gray-500 text-sm menu-remove-button";
+    // 메뉴 삭제 버튼을 이용하여 메뉴를 삭제할 수 있다
+    menuItemDeleteButton.onclick = (e) => deleteMenu(e);
+    menuItemDeleteButton.appendChild(document.createTextNode("삭제"));
+    menuItem.appendChild(menuItemName);
+    menuItem.appendChild(menuItemSoldOutButton);
+    menuItem.appendChild(menuItemUpdateButton);
+    menuItem.appendChild(menuItemDeleteButton);
+    $menuList.appendChild(menuItem);
+  };
+
+  const saveMenu = () => {
+    localStorage.setItem("espresso", JSON.stringify(menuData));
+  };
+
+  const loadMenu = () => {
+    const savedMenuData = localStorage.getItem("espresso");
+    if (savedMenuData !== null) {
+      const parsedData = JSON.parse(savedMenuData);
+      parsedData.forEach(drawMenu);
+      menuData = parsedData;
+    }
+  };
+
+  const createMenu = (e) => {
+    e.preventDefault();
+    // 사용자 입력값이 빈 값이라면 추가되지 않는다
+    if ($menuInput.value === "") return alert(MESSAGE.ALERT_CREATE);
+    const newMenu = {
+      id: `${Date.now()}`,
+      name: $menuInput.value,
+      isSoldOut: false,
+    };
+    drawMenu(newMenu);
+    menuData.push(newMenu);
+    saveMenu();
+    countMenu();
+    // 메뉴가 추가되고 나면, input은 빈 값으로 초기화한다
+    $menuForm.reset();
+  };
+
+  const updateMenu = ({ target }) => {
+    const currentName = target.parentElement.querySelector(".menu-name");
```

Comment:
```suggestion
    const $currentName = target.parentElement.querySelector(".menu-name");
```

윗부분에는 dom에 $ prefix 를 붙이신 것 같은데 여기는 빼먹으신 것 같아요 😀

Reply:
이런 세세한 부분까지 봐주시고 너무 감사합니다 ㅎㅎ